heron/stmgr/src/cpp/manager/stmgr-server.cpp

```diff
@@ -667,5 +682,117 @@ void StMgrServer::AttemptStopBackPressureFromSpouts() {
   }
 }
 
+void StMgrServer::InitiateStatefulCheckpoint(const sp_string& _checkpoint_tag) {
+  for (auto iter = instance_info_.begin(); iter != instance_info_.end(); ++iter) {
+    if (iter->second->is_local_spout() && iter->second->conn_) {
```

Comment:
What is a local spout? Is that a marker on spout instances that are on this container?